src/Data/Yaml.php

```diff
@@ -42,16 +44,15 @@ public static function decode($string): array
 			throw new InvalidArgumentException('Invalid YAML data; please pass a string');
 		}
 
-		// remove BOM
-		$string = str_replace("\xEF\xBB\xBF", '', $string);
-		$result = Spyc::YAMLLoadString($string);
+		return match (static::handler()) {
+			'symfony' => YamlSymfony::decode($string),
+			default   => YamlSpyc::decode($string)
+		};
+	}
 
-		if (is_array($result) === true) {
-			return $result;
-		}
+	public static function handler(): string
```

Comment:
I think it would be good to also have a doc block here. It's not immediately obvious that the return value will be just the handler type without looking at the code. Of course the return type already tells us that it's not an object like with `Data::handler()`, but a string could still be anything.

Also I'd mark it `@internal` as we will remove this method again once we remove `spyc` support.

```suggestion
	/**
	 * Returns the configured YAML library (`'symfony'` or `'spyc'`)
	 * @internal
	 */ 
	public static function handler(): string
```